### PR TITLE
Add instructions to set up a Windows development environment

### PR DESCRIPTION
**Pull request by ludelafo**

### CODE REVIEW

File: 01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+>
+> All Linux distributions come with a package manager that is used to install,
+> update, and remove packages. The package manager for Ubuntu is called `apt`.
+>
+> The package manager uses a package list to know which packages are available
+> for installation. You can then install any package from the package list using
+> the package manager.
+>
+> If you are interested to have a package manager for Windows, you can check
+> [Chocolatey](https://chocolatey.org/) or [Scoop](https://scoop.sh/).
```
Comment (hadrylouis) [inline 01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md:332]: I would also add [winget](https://learn.microsoft.com/fr-fr/windows/package-manager/) , as it is the package management solution created by Microsoft.

File: 01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+Once you create a new project, IntelliJ IDEA should automatically detect the
+Java and Maven installations in WSL.
+
+Create a new project in your WSL home directory using the same instructions as
+in the course material.
+
+> [!CAUTION]
+>
+> Always create your projects in your WSL home directory to avoid file
+> permission issues!
```
Comment (hadrylouis) [inline 01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md:487]: That's a very good point. 

Perhaps I'll add some further explanations or an extra chapter with the creation of a `~/devs/dai` folder, for example, and then the creation of a Java / Maven Hello world project in order to : 

1. Make sure everything works.
2. Force students to create their projects in a specific dev folder with a default structure for all future HEIG-VD courses. I think it's a good thing to be fairly strict on this point, in order to teach good practices right from the start and prevent students from storing their sources on OneDrive, Dropbox or anywhere else on the OS.

We might also add some examples of the basic commands (ls, cd etc...) to move inside the file tree and then compile and launch the hello world project with java, javac

**Review by hadrylouis — APPROVED** [review]

File: 01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md (hunk 1 above)
Comment (ludelafo) [reply 01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md]: Good point, I have added it in my last commit.

File: 01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md (hunk 2 above)
Comment (ludelafo) [reply 01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md]: Very valid points. I have added some instructions to the guide and added a separated "Considerations for a development environment" guide to address some of these issues.

File: 01-introduction-and-course-organization/CONSIDERATIONS_FOR_A_DEVELOPMENT_ENVIRONMENT.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# Considerations for a development environment
+
+## Do not store your projets' sources on a cloud provider
+
+We highly recommend to **never store projects' sources your files on
```
Comment (hadrylouis) [inline 01-introduction-and-course-organization/CONSIDERATIONS_FOR_A_DEVELOPMENT_ENVIRONMENT.md:5]: maybe : to **never store your projects' sources files on...

Comment (ludelafo) [reply 01-introduction-and-course-organization/CONSIDERATIONS_FOR_A_DEVELOPMENT_ENVIRONMENT.md]: Thanks! It's fixed.

File: 01-introduction-and-course-organization/CONSIDERATIONS_FOR_A_DEVELOPMENT_ENVIRONMENT.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+## Do not store your projets' sources on a cloud provider
+
+We highly recommend to **never store your projects' sources your files on
+OneDrive/Google Drive/Dropbox/etc.**. When you start to develop professionally,
+your IDE and tools can generate a large amount of temporary files (cache files,
+logs, compiled files, etc.). These files are often small but numerous and can be
+generated every time you compile your project. As these cloud services will try
+to keep up to synchronize these files, it can quickly use all your computer's
+resources and can slow down your computer/develoment environment. As Git is
```
Comment (rmarquis) [inline 01-introduction-and-course-organization/CONSIDERATIONS_FOR_A_DEVELOPMENT_ENVIRONMENT.md:18]: typo: `develoment`

File: 01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md [hunk 5]
```diff
@@ -0,0 +1,684 @@
...
+version.
+
+### Check for Windows updates
+
+You can check if your Windows installation is up to date by running the
+following command in a PowerShell terminal:
+
+```powershell
+# Check if your Windows installation is up to date
+sconfig
```
Comment (rmarquis) [inline 01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md:639]: I believe the `sconfig` command is only available on Windows Server version (and maybe Enterprise). It doesn't seem to be available on Windows Pro/Home.

Comment (ludelafo) [reply 01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md]: You're right. I've updated the document to reflect this (I wasn't able to find a quick command to check that aspect and other part of the guide were already mentioning to go through the GUI).

File: 01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md [hunk 6]
```diff
@@ -0,0 +1,684 @@
...
+```
+
+### Check your Windows features
+
+You can check if WSL is available on your Windows installation by running the
+following command in a PowerShell terminal:
+
+```powershell
+# Check if WSL is available
+dism.exe /online /get-features | Select-String "Microsoft-Windows-Subsystem-Linux"
```
[the remark this answers is not in the document]
Comment (ludelafo) [reply 01-introduction-and-course-organization/SET_UP_A_WINDOWS_DEVELOPMENT_ENVIRONMENT.md]: Thank you, I've updated the guide to mention to run the PowerShell terminal as administrator.

**Pull request merged**